### ISSUE TITLE
borg recreate --timestamp is a no-op

### ISSUE THREAD

**Issue by horazont** [borgbackup/borg#4745]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

BUG (either docs or borg-recreate)

#### Your borg version (borg -V).

```
$ borg -V
borg 1.1.10
```

#### Operating system (distribution) and version.

4.19.0-5-amd64, Debian testing

#### Hardware / network configuration, and filesystems used.

tmpfs

#### How much data is handled by borg?

few MB

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

```
$ borg recreate --timestamp '2019-01-01T00:00:00' --comment 're-created for fun' testrepo::archive1
```

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

Yes, 100%. Assuming you have some stuff to play with in ``./testcontent``.

```console
$ borg init -e none testrepo
$ borg create testrepo::archive1 testcontent
$ borg info testrepo::archive1
Archive name: archive1
Archive fingerprint: fc4f5feb66c314073d124acd10c856b153f077e5865fd655b23423d21b6227ac
Comment:
Hostname: sinistra
Username: horazont
Time (start): Sun, 2019-09-01 17:02:18
Time (end): Sun, 2019-09-01 17:02:18
Duration: 0.03 seconds
Number of files: 2
Command line: /usr/bin/borg create testrepo::archive1 testcontent
Utilization of maximum supported archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:                3.75 MB              3.76 MB              3.76 MB
All archives:                3.75 MB              3.76 MB              3.76 MB

                       Unique chunks         Total chunks
Chunk index:                       5                    5
$ borg recreate --timestamp '2019-01-01T00:00:00' --comment 're-created for fun' 
recreate is an experimental feature.
Type 'YES' if you understand this and want to continue: YES
$ borg info testrepo::archive1
Archive name: archive1
Archive fingerprint: f317eeb7a04e2fa1f06b3d0e3aa919042cb6730fc1d913d348f08e768d800314
Comment: re-created for fun
Hostname: sinistra
Username: horazont
Time (start): Sun, 2019-09-01 17:02:18
Time (end): Sun, 2019-09-01 17:02:18
Duration: 0.03 seconds
Number of files: 2
Command line: /usr/bin/borg create testrepo::archive1 testcontent
Utilization of maximum supported archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:                3.75 MB              3.76 MB              3.76 MB
All archives:                3.75 MB              3.76 MB              3.76 MB

                       Unique chunks         Total chunks
Chunk index:                       5                    5
```

I expect the timestamp in the second borg info call to be changed after borg recreate, but that is not happening.

Bonus:
```console
$ borg recreate --timestamp '2019-01-01T00:00:00' testrepo::archive1
recreate is an experimental feature.
Type 'YES' if you understand this and want to continue: YES
Nothing to do. Archive was not processed.
Specify at least one pattern, PATH, --comment, re-compression or re-chunking option.
```

It appears the code knows about this, too.

I noticed that args.timestamp is not used at all here: https://github.com/borgbackup/borg/blob/master/src/borg/archiver.py#L1538

**Comment by Gu1nness** [comment]
This clearly questions the behavior expected of the `--timestamp` option.
For now, it is supposed to set the archive crate date/time, which can be either the starting time of the archive creation date start and end (need some small computations, but easy), or only the start (then, the end would be the current timestamp ?).

I'll write some proofs of concepts of both waiting for an answer for this.

**Comment by jwestapplause** [comment]
Is this issue still open or has a fix been accepted?

**Comment by jameshwest** [comment]
Is this issue still open or has a fix been accepted?

**Comment by Gu1nness** [comment]
The fix is still being developed, see the PR #4815 . I have pushed a commit yesterday.
